index.js

```diff
@@ -168,6 +168,18 @@ function session(options){
     // req.secret is passed from the cookie parser middleware
     var secrets = secret || [req.secret];
 
+    // Since we execute the functions (if they are a secet) we need
+    // to make a copy so that we don't overwrite the original function
+    var processedSecrets = [];
+
+    for (var i = 0; i < secrets.length; i++) {
```

Comment:
Both sound good to me. I only added support for the function in an array to try and be consistent.